e2e/specs/ward-admission.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { expect } from '@playwright/test';
+import { test } from '../core';
+import {
+  endVisit,
+  startVisit,
+  changeToWardLocation,
+  generateRandomPatient,
+  getProvider,
+  generateWardAdmission,
+  deletePatient,
+} from '../commands';
+import { type Visit } from '@openmrs/esm-framework';
+import { type Patient, type Encounter, type Provider, type Bed, type BedType } from '../commands/types';
+import {
+  deleteBed,
+  dischargePatientFromBed,
+  generateBedType,
+  generateRandomBed,
+  retireBedType,
+} from '../commands/bed-operations';
+import { WardPage } from '../pages';
+
+let visit: Visit;
+let wardPatient: Patient;
+let encounter: Encounter;
+let provider: Provider;
+let bed: Bed;
+let bedtype: BedType;
+
+test.beforeEach(async ({ api }) => {
+  await changeToWardLocation(api);
+  bedtype = await generateBedType(api);
+  bed = await generateRandomBed(api, bedtype);
+  provider = await getProvider(api);
+  wardPatient = await generateRandomPatient(api, process.env.E2E_WARD_LOCATION_UUID);
+  visit = await startVisit(api, wardPatient.uuid, process.env.E2E_WARD_LOCATION_UUID);
+  encounter = await generateWardAdmission(api, provider.uuid, wardPatient.uuid);
+});
+
+test('Confirming patient is admitted to ward', async ({ page }) => {
```

Comment:
This should be a bit more descriptive of what the overall test does, similar to the other specs
```suggestion
test('Admit a patient to a ward from the admission requests', async ({ page }) => {
```